systori/apps/task/serializers.py

```diff
@@ -1,58 +1,17 @@
+from collections import OrderedDict
+
+
+from django.db import models
+
 from rest_framework import serializers
+from rest_framework.fields import SkipField
+
 from rest_framework.generics import get_object_or_404
-from .models import Job, Group, Task, LineItem
-
-
-class RecursiveField(serializers.Field):
-
-    PROXIED_ATTRS = (
-        # methods
-        "get_value",
-        "get_initial",
-        "run_validation",
-        "get_attribute",
-        "to_representation",
-        # attributes
-        "field_name",
-        "source",
-        "read_only",
-        "default",
-        "source_attrs",
-        "write_only",
-    )
-
-    def __init__(self, **kwargs):
-        self._proxied = None
-        super().__init__(**kwargs)
-
-    def bind(self, field_name, parent):
-        self.bind_args = (field_name, parent)
-
-    @property
-    def proxied(self):
-        if not self._proxied:
-            if self.bind_args:
-                field_name, parent = self.bind_args
-                if hasattr(parent, "child") and parent.child is self:
-                    # RecursiveField nested inside of a ListField
-                    parent_class = parent.parent.__class__
-                else:
-                    # RecursiveField directly inside a Serializer
-                    parent_class = parent.__class__
-                proxied = parent_class(many=True, required=False)
```

Comment:
Here `many=True` makes the recursive field always convert itself to `ListSerializer(child=GroupSerializer())`. But the problem is that if we use a `ListField(child=RecursiveField())` then the resulting serializer becomes `ListSerializer(child=ListSerializer(child=GroupSerializer()))` Which is wrong.
I suspect this was probably done because we only ever use `RecursiveField `in the `GroupSerializer`. And inside of that it's _always_ a list of group instances.

Anyhow, the `RecursiveField` from `djangorestframwork-recursive` works nicely

Reply:
ok perfect